File: engine/src/main/java/de/hanno/hpengine/engine/graphics/state/DirectionalLightState.java
package de.hanno.hpengine.engine.graphics.state;

import org.lwjgl.BufferUtils;
import org.lwjgl.util.vector.Vector3f;

import java.nio.FloatBuffer;

public class DirectionalLightState {
    public FloatBuffer directionalLightViewMatrixAsBuffer = BufferUtils.createFloatBuffer(16);
    public FloatBuffer directionalLightProjectionMatrixAsBuffer = BufferUtils.createFloatBuffer(16);
    public FloatBuffer directionalLightViewProjectionMatrixAsBuffer = BufferUtils.createFloatBuffer(16);
    public Vector3f directionalLightDirection = new Vector3f();
    public Vector3f directionalLightColor = new Vector3f();
    public float directionalLightScatterFactor;

    public DirectionalLightState() {
    }
}

File: engine/src/main/java/de/hanno/hpengine/engine/graphics/state/DirectionalLightState.kt
package de.hanno.hpengine.engine.graphics.state

import de.hanno.hpengine.engine.graphics.HpMatrix
import de.hanno.hpengine.engine.scene.HpVector3f
import de.hanno.struct.Struct
import de.hanno.struct.Structable

class DirectionalLightState(parent: Structable? = null) : Struct(parent) {
    val viewMatrix by HpMatrix(this)
    val projectionMatrix by HpMatrix(this)
    val viewProjectionMatrix by HpMatrix(this)
    val color by HpVector3f(this)
    val dummy by 0F
    val direction by HpVector3f(this)
    var scatterFactor by 0F
    var shadowMapHandle by 0L
    var shadowMapId by 0
    val dummy0 by 0F
}